runtime/src/rpc.rs

```diff
@@ -9,50 +11,24 @@ use crate::{BTC_RELAY_MODULE, STABLE_BITCOIN_CONFIRMATIONS, STABLE_PARACHAIN_CON
 use async_trait::async_trait;
 use codec::Encode;
 use futures::{future::join_all, stream::StreamExt, FutureExt, SinkExt};
-use jsonrpsee::types::to_json_value;
+use jsonrpsee::core::to_json_value;
 use module_oracle_rpc_runtime_api::BalanceWrapper;
-use sp_runtime::create_runtime_str;
-use sp_version::RuntimeVersion;
-use std::{borrow::Cow, collections::BTreeSet, future::Future, sync::Arc, time::Duration};
+use std::{collections::BTreeSet, future::Future, sync::Arc, time::Duration};
 use subxt::{
-    sp_runtime::DispatchError, Client as SubxtClient, ClientBuilder as SubxtClientBuilder, Event, EventSubscription,
-    EventsDecoder, Metadata, RpcClient, RuntimeError as SubxtRuntimeError, Signer,
+    BasicError, Client as SubxtClient, ClientBuilder as SubxtClientBuilder, DefaultExtra, Event, EventSubscription,
+    EventsDecoder, Metadata, RpcClient, Signer, TransactionEvents, TransactionProgress,
 };
 use tokio::{sync::RwLock, time::sleep};
 
 const DEFAULT_COLLATERAL_CURRENCY: CurrencyId = Token(DOT);
 
 cfg_if::cfg_if! {
     if #[cfg(feature = "standalone-metadata")] {
-        const DEFAULT_RUNTIME_VERSION: RuntimeVersion = RuntimeVersion {
-            spec_name: create_runtime_str!("interbtc-standalone"),
-            impl_name: create_runtime_str!("interbtc-standalone"),
-            authoring_version: 1,
-            spec_version: 1,
-            impl_version: 1,
-            transaction_version: 1,
-            apis: Cow::Owned(vec![]),
-        };
+        const DEFAULT_SPEC_VERSION: u32 = 1;
     } else if #[cfg(feature = "parachain-metadata-kintsugi")] {
-        const DEFAULT_RUNTIME_VERSION: RuntimeVersion = RuntimeVersion {
-            spec_name: create_runtime_str!("kintsugi-parachain"),
-            impl_name: create_runtime_str!("kintsugi-parachain"),
-            authoring_version: 1,
-            spec_version: 9,
-            impl_version: 1,
-            transaction_version: 2,
-            apis: Cow::Owned(vec![]),
-        };
+        const DEFAULT_SPEC_VERSION: u32 = 9;
     } else if #[cfg(feature = "parachain-metadata-testnet")] {
-        const DEFAULT_RUNTIME_VERSION: RuntimeVersion = RuntimeVersion {
-            spec_name: create_runtime_str!("testnet-parachain"),
-            impl_name: create_runtime_str!("testnet-parachain"),
-            authoring_version: 1,
-            spec_version: 0,
-            impl_version: 1,
-            transaction_version: 0,
-            apis: Cow::Owned(vec![]),
-        };
+        const DEFAULT_SPEC_VERSION: u32 = 0;
```

Comment:
Why move away from the runtime compatibility check?

Reply:
`subxt` has redefined [RuntimeVersion](https://github.com/paritytech/subxt/blob/a7c54f74127b58dd0886b9f42b1f7318dfbc51f4/subxt/src/rpc.rs#L177-L198).